### PR TITLE
Specify a context when deploying oracles, add 3rd mainnet cluster

### PR DESCRIPTION
**Pull request by tkporter**
### Description

This does away with the "primary" and "secondary" terminology when referring to an oracle cluster, and instead introduces "contexts" for oracle deployments. An oracle context can really have any value, but in practice it should be: `<cloud provider>-<region>`, eg `azure-westus2` or `azure-westeurope`. This is what should be passed into celotool, for example:
```
celotooljs deploy initial oracle -e oracledev --context azure-eastus
```

Behind the scenes, this context is coerced into being all caps and using underscores instead of dashes. Env variables for a an oracle context are expected to be present and have a prefix of the form `ORACLE_<context in caps and underscores>_.*` (eg `ORACLE_AZURE_EASTUS_AZURE_SUBSCRIPTION_ID`). To do this, `DynamicEnvVar`s are introduced, which are a way of dynamically generating an environment variable name. For example:
```
ORACLE_AZURE_SUBSCRIPTION_ID = 'ORACLE_{{ oracleContext }}_AZURE_SUBSCRIPTION_ID',
``` 

In this case, `{{ oracleContext }}` is replaced at runtime with the value of whatever the oracleContext is.

This also moves oracle full node env vars to be context-specific, so that one context can have more/less full nodes or more/less storage per node.

### Other changes

* Adds a 3rd cluster in Azure that has full nodes constantly running as an emergency backup.
* Adds the ability to optionally specify which resource group a key vault is located in: `<address>:<key vault name>:<resource group (optional)>`

### Tested

Deployed!

### Related issues

n/a

### Backwards compatibility

Env variables must be updated

### CODE REVIEW

File: packages/celotool/src/lib/oracle.ts [hunk 1]
```diff
@@ -341,45 +332,83 @@ function getOracleAzureIdentityName(keyVaultName: string, address: string) {
...
+  {})
 }
 
 /**
  * Switches to the AKS cluster associated with the given context
  */
-export function switchToAzureContextCluster(celoEnv: string, context: OracleAzureContext) {
-  const azureClusterConfig = getAzureClusterConfig(context)
+export function switchToAzureContextCluster(celoEnv: string, oracleContext: string) {
+  const validOracleContexts = fetchEnv(envVar.ORACLE_CONTEXTS)
```
Comment (jasonrudianto1) [inline packages/celotool/src/lib/oracle.ts]: I like having this check! I can see this being used for a switchToAwsContextCluster function. Perhaps lines 381-386 can be made into a new "isValidOracleContext" function for reusability and possibly better readability?

**Review by jasonrudianto1 — APPROVED** [review]
Looks great to me!

File: packages/celotool/src/lib/env-utils.ts [hunk 2]
```diff
@@ -216,6 +218,16 @@ export function isValidCeloEnv(celoEnv: string) {
   return new RegExp('^[a-z][a-z0-9]*$').test(celoEnv)
 }
 
+/**
+ * Replaces a dynamic env var's template strings with values from an object.
+ * Returns the name of the env var.
+ */
+export function getDynamicEnvVarName(dynamicEnvVar: DynamicEnvVar, templateValues: any) {
+  return Object.keys(templateValues).reduce((agg: string, templateKey: string) => {
```
Comment (codyborn) [inline packages/celotool/src/lib/env-utils.ts:234]: It took me a min to figure out what was going on here 😄 Looks good! Maybe you can include usage patterns in the comments for examples.

File: packages/celotool/src/lib/oracle.ts [hunk 3]
```diff
@@ -47,40 +42,24 @@ interface OracleConfig {
...
-    clusterName: envVar.ORACLE_SECONDARY_AZURE_KUBERNETES_CLUSTER_NAME,
-  },
+const oracleContextAzureClusterConfigDynamicEnvVars: { [k in keyof AzureClusterConfig]: DynamicEnvVar } = {
+  subscriptionId: DynamicEnvVar.ORACLE_AZURE_SUBSCRIPTION_ID,
+  tenantId: DynamicEnvVar.ORACLE_AZURE_TENANT_ID,
+  resourceGroup: DynamicEnvVar.ORACLE_AZURE_KUBERNETES_RESOURCE_GROUP,
+  clusterName: DynamicEnvVar.ORACLE_KUBERNETES_CLUSTER_NAME,
+}
+
+interface OracleIdentityConfig {
```
Comment (codyborn) [inline packages/celotool/src/lib/oracle.ts]: Given that only one of these will be defined, perhaps it's cleaner to define two separate objects.
```
interface KeyVaultIdentityConfig {
  addressAzureKeyVaults: string
}
interface MnemonicIdentityConfig {
  addressesFromMnemonicCount: string
}
type OracleIdentityConfig = KeyVaultIdentityConfig | MnemonicIdentityConfig
```

**Review by codyborn — APPROVED** [review]
Looks good!

**Pull request merged**